### PR TITLE
MM-38140 - show only enterprise for professional upgrade

### PR DESCRIPTION
**Pull request by pablovelezvidal**
#### Summary
When a customer is upgrading to Enterprise from professional, we want to only show the Enterprise option.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-38140

#### Related Pull Requests
none

#### Screenshots
|  before  |  after  |
|----|----|
| 
Captura de pantalla 2021-09-14 a las 9 27 30
 | 
Captura de pantalla 2021-09-14 a las 9 27 09
 |

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by AGMETEOR — APPROVED** [review]

**Comment by AGMETEOR** [comment]
@neallred I want to approve but I feel like I should wait until all the PM feedback is completed since there would be code changes.

**Review by stevemudie — APPROVED** [review]
All set @pablovelezvidal 

After the initial purchase of Professional all I could see when upgrading from there was Enterprise as expected.

File: components/purchase_modal/purchase_modal.tsx [hunk 1]
```diff
@@ -527,7 +531,7 @@ export default class PurchaseModal extends React.PureComponent<Props, State> {
                 </div>
                 <div className='RHS'>
                     <div className='price-container'>
-                        {(this.props.products && Object.keys(this.props.products).length > 1) &&
+                        {(this.props.products && Object.keys(this.props.products).length > 1 && this.state.currentProduct?.sku !== 'cloud-professional') &&
```
Comment (marianunez) [inline components/purchase_modal/purchase_modal.tsx:534]: Why do we need to hardcode the condition here to check for product name? If there is more than 1 product then it should have the compare plans. In the case of current sku is `Starter` there should be more than one product so it should show the compare plans and in the case of `Professional` it should only have 1 product and not show this

File: components/purchase_modal/purchase_modal.tsx [hunk 2]
```diff
@@ -224,6 +224,10 @@ export default class PurchaseModal extends React.PureComponent<Props, State> {
             if (products[key].billing_scheme === BillingSchemes.FLAT_FEE) {
                 flatFeeProducts.push(tempEl);
             } else {
+                // only show enterprise to customers upgrading from professional
+                if (currentProduct.sku === 'cloud-professional') {
+                    return;
+                }
```
Comment (marianunez) [inline components/purchase_modal/purchase_modal.tsx:230]: Why filter here for an specific product if it filters by product price below?

**Review by marianunez — CHANGES_REQUESTED** [review]
We should avoid putting conditions for specific skus - let's review the logic we had before for ordering/filtering by product price

File: components/purchase_modal/purchase_modal.tsx (hunk 1 above)
Comment (pablovelezvidal) [reply components/purchase_modal/purchase_modal.tsx:534]: ok

File: components/purchase_modal/purchase_modal.tsx (hunk 2 above)
Comment (pablovelezvidal) [reply components/purchase_modal/purchase_modal.tsx:230]: ok

**Review by stevemudie — APPROVED** [review]
Stepped through from Starter to Professional and when I went to upgrade to Enterprise it was the only option displayed as expected.

**Review by marianunez — APPROVED** [review]
Thanks @pablovelezvidal!

**Review by AGMETEOR — APPROVED** [review]

**Review by annelieseklein — APPROVED** [review]

**Pull request merged**